### DISCUSSION TITLE
Install dependencies from metric_providers individually in install script

### DISCUSSION THREAD

**Discussion by ArneTR** [green-coding-solutions/green-metrics-tool#492] [Ideas]
This idea came originally from @ribalba ❤️ 

In the https://github.com/green-coding-berlin/green-metrics-tool/pull/488 it became clear that metric providers need their own checks and they should also have their own install routines.

This makes them even more decoupled and can make the install more concise.

The proposal is:
- Have an install.sh in every metric provider directory
- Asking in the install_[mac,linux].sh for every found install.sh script in a metric_provider subfolder if the provider shall be installed
  + This way every provider can be easily selected / deselected and only the most necessary stuff is installed.
  + Also compilation with GCC should only happen if the user wants that. This way we can then only use the needed apt-install packages we need
- we can also think of unbundling the XGBoost reporter repository as a git submodule and just download the folder on request. Updating is then tied to the install script and not anymore to a git pull ... TBD

This makes it way clearer for the user WHICH reporters do even exist in the tool. During the installation they get a mini-summary what the reporter does and when it should be installed.

Also then later the documentation has to be touched and we should move the install instructions out of the install documentation itself but to the metric providers.

Your comments please :) @ribalba @dan-mm

**Comment by ribalba** [comment]
I like the idea of not installing everything per default. We could add the content to the makefile and have a `make install` which installs everything and also builds the binary. 

Also the XGBoost providers needs to install pip packages which I have forgotten multiple times.
